README.md

```diff
@@ -1,2 +1,115 @@
+
+## Explicit Instrumentation
+
+One can explicitly add instrumentation to a library or an application. Currently the only hooks intended for explicit programmer use are the Region related hooks. These use a push/pop model to mark coarser regions in your code.
+
+```c++
+void foo() {
+   Kokkos::Profiling::pushRegion("foo");
+   bar();
+   stool();
+   Kokkos::Profiling::popRegion();
+}
+```
+
+## Tools
+
+### Utilities
+
++ **[[KernelFilter|KernelFilter]]:**
+
+   A tool which is used in conjunction with analysis tools, to restrict them to a subset of the application.
+
+### Memory Analysis
++ **[[MemoryHighWater|MemoryHighWater]]:**
+
+    Outputs high water mark of memory usage of the application.
+
++ **[[MemoryUsage|MemoryUsage]]:**
+
+    Generates a per Memory Space timeline of memory utilization. 
+
++ **[[MemoryEvents|MemoryEvents]]:** 
+
+    Tool to track memory events such as allocation and deallocation. It also provides the information of the MemoryUsage tool.
+
+### Kernel Inspection
++ **[[SimpleKernelTimer|SimpleKernelTimer]]:**
+
+    Captures Basic Timing information for Kernels.
+
++ **[[KernelLogger|KernelLogger]]:**
+
+    Prints Kernel and Region events during runtime.
+
+### 3rd Party Profiling Tool Hooks
++ **[[VTuneConnector|VTuneConnector]]:**
+    
+    Provides Kernel Names to VTune, so that analysis can be performed on a per kernel base.
+
++ **[[VTuneFocusedConnector|VTuneFocusedConnector]]:**
+    
+    Like VTuneConnector but turns profiling off outside of kernels. Should be used in conjunction with the KernelFilter tool. 
+
++ **[[Timemory|Timemory]]:**
+
+    Modular connector for accumulating timing, memory usage, hardware counters, and other various metrics.
+    Supports controlling VTune, CUDA profilers, and TAU + kernel name forwarding to VTune, NVTX, TAU,
+    Caliper, and LIKWID.
+
+    #####  If you need to write your own plug-in, this provides a straight-forward API to writing the plug-in.
+
+    Defining a timemory component will enable your plug-in to output to stdout, text, and JSON, 
+    accumulate statistics, and utilize various portable function calls for common needs w.r.t. timers,
+    resource usage, etc. 
+
+
+# Building Kokkos-tools
+
+Use either cmake or provided Makefile within each tools directory. 
+
+Building with Makefiles is currently recommended. 
+
+# Running a Kokkos-based Application with a tool
+
+
+
+Given your tool shared library <name_of_tool_shared_library>.so (which contains kokkos profiling callback functions) and an application executable called yourApplication.exe, type 
+
+`export KOKKOS_PROFILE_LIBRARY=${YOUR_KOKKOS_TOOLS_DIR}/<name_of_tool_shared_lib>; ./yourApplication.exe`
+
+
+
+
+
+# Documentation
+
+Further information and tutorials can be found here: https://github.com/kokkos/kokkos-tutorials/blob/main/LectureSeries/KokkosTutorial_07_Tools.pdf
```

Comment:
Fine but in the long run we should refer to the wiki page that has a link to the lecture recording on YouTube as well as these slides

Reply:
Yes, will note that